src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2168,6 +2181,107 @@ private CallResult ExecuteCall<TTracingInstructions>(EvmState vmState, ReadOnlyM
+
+        if (!stack.PopUInt256(out UInt256 offset)) return EvmExceptionType.StackUnderflow;
+        if (!stack.PopUInt256(out UInt256 length)) return EvmExceptionType.StackUnderflow;
+
+        if (_state.IsContract(authority))
+        {
+            vmState.Authorized = null;
+            stack.PushUInt256(0);
+            return EvmExceptionType.None;
+        }
+
+        gasAvailable -= GasCostOf.Auth;
+
+        if (!UpdateMemoryCost(vmState, ref gasAvailable, offset, length))
+            return EvmExceptionType.OutOfGas;
+
+        if (!ChargeAccountAccessGas(ref gasAvailable, vmState, authority, spec))
+            return EvmExceptionType.OutOfGas;
+
+        ReadOnlySpan<byte> memData = vmState.Memory.Load(in offset, length).Span;
+
+        byte yParity = 0;
+        Span<byte> sigData = stackalloc byte[64];
+        Span<byte> commit = stackalloc byte[32];
+        //SkipLocalsInit flag is active so we have to iterate all data
+        for (int i = 0; i < 97; i++)
+        {
+            byte data = 0;
+            if (i < length)
+                data = memData[i];
+            switch (i)
+            {
+                case 0:
+                    yParity = data;
+                    break;
+                case >= 1 and <= 64:
+                    sigData[i - 1] = data;
+                    break;
+                default:
+                    commit[i - 65] = data;
+                    break;
+            }
+        }
+        Signature signature = new(sigData, yParity);
+
+        Address recovered = null;
+        if (signature.V == 27 || signature.V == 28)
+        {
+            //TODO is ExecutingAccount correct when DELEGATECALL and CALLCODE?
+            recovered = TryRecoverSigner(signature, vmState.Env.ExecutingAccount, authority, commit);
+        }
+
+        if (recovered == authority)
+        {
+            stack.PushUInt256(1);
+            vmState.Authorized = authority;
+        }
+        else
+        {
+            stack.PushUInt256(0);
+            vmState.Authorized = null;
+        }
+        return EvmExceptionType.None;
+    }
+
+    private Address? TryRecoverSigner(
+        Signature signature,
+        Address invoker,
+        Address authority,
+        ReadOnlySpan<byte> commit)
+    {
+        byte[] chainId = _chainId.PadLeft(32);
+        UInt256 nonce = _state.GetNonce(authority);
+        byte[] invokerAddress = invoker.Bytes;
+        Span<byte> msg = stackalloc byte[1 + 32 * 4];
+        msg[0] = Eip3074Constants.AuthMagic;
+        for (int i = 0; i < 32; i++)
+        {
+            int offset = i + 1;
+            msg[offset] = chainId[i];
+            msg[32 + 32 - i] = (byte)(nonce[i / 8] >> (8 * (i % 8)));
+
+            if (i < 12)
+                msg[offset + 64] = 0;
+            else if (i - 12 < invokerAddress.Length)
+                msg[offset + 64] = invokerAddress[i - 12];
+
+            if (i < commit.Length)
+                msg[offset + 96] = commit[i];
+        }
```

Comment:
This is probably suboptimal, you can write chainid directly to message rather than creating byte array, similar you can probably SIMD and not for some other bytes.